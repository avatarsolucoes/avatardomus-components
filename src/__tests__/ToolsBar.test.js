/* eslint-disable prettier/prettier */
import React from 'react';
import renderer from 'react-test-renderer';
// import { renderHook } from '@testing-library/react-hooks';

import { shallow, mount } from 'enzyme';

import { ToolsBar, ToolButton, ToolItem, ToolLink, ItemTitle, ItemButtons, ToolSelectItem } from '..';

const daysOptions = [
  { id: 10, title: '10 dias' },
  { id: 30, title: '30 dias' },
  { id: 60, title: '60 dias' },
  { id: 90, title: '90 dias' },
];

describe('TableDiv', () => {
  it('is truthy', () => {
    expect(ToolsBar).toBeTruthy();
    expect(ToolButton).toBeTruthy();
    expect(ToolItem).toBeTruthy();
    expect(ToolLink).toBeTruthy();
  });

  it('Test ToolSelectItem', () => {
    const wrapper = mount(
      <ToolsBar line>
        <ItemTitle>My Test</ItemTitle>
        <ToolSelectItem defaultOptions={daysOptions} name="lastDays" placeholder="test" defaultValue={10} />
      </ToolsBar>
    );
    wrapper
      .find('option')
      .at(0)
      .instance().selected = false;
    wrapper
      .find('option')
      .at(1)
      .instance().selected = true;
    wrapper.find('select').simulate('change', { target: { value: '30' } }).update();
    expect(wrapper.html()).toMatchSnapshot();

    const tree = renderer.create(<ToolSelectItem defaultOptions={daysOptions} name="lastDays" value={10} />);
    tree.update();
  });

  it('Test TableDiv renderer whith children', () => {
    const wrapper = shallow(
        <ToolsBar line>
          <ItemTitle text="My test" />
          <ItemTitle>Teste title</ItemTitle>
          <ItemButtons>
            <ToolButton icon="del" value={1} disabled />
            <ToolButton icon="update" />
            <ToolLink icon="add" value="https://avatarsolucoesdigitais.com.br" />
            <ToolLink icon="add" value="https://avatarsolucoesdigitais.com.br" disabled />
          </ItemButtons>
        </ToolsBar>
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
});
