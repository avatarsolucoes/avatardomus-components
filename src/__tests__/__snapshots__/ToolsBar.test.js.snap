// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableDiv Test TableDiv renderer whith children 1`] = `"<div class=\\"undefined toolsbar line\\"><div class=\\"toolsbar-item toolsbar-title\\"><h3>My test</h3></div><div class=\\"toolsbar-item toolsbar-title\\">Teste title</div><div class=\\"toolsbar-item toolsbar-buttons\\"><div class=\\"toolsbar-item\\"><button type=\\"button\\" class=\\"toolsbar-btn hover disabled\\" value=\\"1\\" disabled=\\"\\" style=\\"background-image:url([object Object])\\"></button></div><div class=\\"toolsbar-item\\"><button type=\\"button\\" class=\\"toolsbar-btn hover actived\\" value=\\"0\\" style=\\"background-image:url([object Object]);background-color:#000\\"></button></div><div class=\\"toolsbar-item\\"><button type=\\"button\\" class=\\"toolsbar-btn hover\\" style=\\"background-image:url([object Object])\\"></button></div><div class=\\"toolsbar-item\\"><a href=\\"https://avatarsolucoesdigitais.com.br\\" class=\\"toolsbar-btn hover\\" style=\\"background-image:url([object Object])\\"> </a></div><div class=\\"toolsbar-item\\"><a href=\\"\\" class=\\"undefined toolsbar-btn hover disabled\\" disabled=\\"\\" style=\\"background-image:url([object Object])\\"> </a></div></div></div>"`;

exports[`TableDiv Test ToolSelectItem 1`] = `"<div class=\\"undefined toolsbar line\\"><div class=\\"toolsbar-item toolsbar-title\\">My Test</div><div class=\\"undefined toolsbar-select-item\\"><select name=\\"lastDays\\" class=\\"primary\\" style=\\"padding-right: 2px;\\"><option value=\\"\\" disabled=\\"\\">test</option><option value=\\"10\\">10 dias</option><option value=\\"30\\">30 dias</option><option value=\\"60\\">60 dias</option><option value=\\"90\\">90 dias</option></select></div></div>"`;
